### ISSUE TITLE
Strange behaviour with allOf

### ISSUE THREAD

**Issue by myinsiders** [etorreborre/specs2#319]
The following test is passing:

```List() must contain(allOf(beEqualTo("1234")))```

would this not be expected to fail?

Thx!

**Comment by etorreborre** [comment]
Which version are you using? Can you try the latest one (2.4.15)? (Sorry for the delayed answer, I'm on holidays)

**Comment by myinsiders** [comment]
I just reproed it with 2.4.15:

```
scala> import org.specs2.matcher.MustMatchers._
import org.specs2.matcher.MustMatchers._

scala> List() must contain(allOf(beEqualTo("1234")))
res1: org.specs2.matcher.MatchResult[List[Nothing]] = MatchSuccess(<function0>,<function0>,org.specs2.matcher.MustExpectations$$anon$2@369c91cf)
```

**Comment by etorreborre** [comment]
Thanks for this, I'll check that when I'll get back to my laptop because I'm on holidays right now

**Comment by myinsiders** [comment]
No worries, its very easy to work around. Enjoy your holidays!

—

**Comment by etorreborre** [comment]
This is now published as 2.4.16-SNAPSHOT.
